reactor-netty-http/src/main/java/reactor/netty/http/server/AbstractHttpServerMetricsHandler.java

```diff
@@ -83,94 +88,112 @@ public void channelInactive(ChannelHandlerContext ctx) {
 	@Override
 	@SuppressWarnings("FutureReturnValueIgnored")
 	public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
-		if (msg instanceof HttpResponse) {
-			if (((HttpResponse) msg).status().equals(HttpResponseStatus.CONTINUE)) {
-				//"FutureReturnValueIgnored" this is deliberate
-				ctx.write(msg, promise);
-				return;
-			}
-
-			dataSentTime = System.nanoTime();
-		}
+		try {
+			if (msg instanceof HttpResponse) {
+				if (((HttpResponse) msg).status().equals(HttpResponseStatus.CONTINUE)) {
+					return;
+				}
 
-		if (msg instanceof ByteBufHolder) {
-			dataSent += ((ByteBufHolder) msg).content().readableBytes();
-		}
-		else if (msg instanceof ByteBuf) {
-			dataSent += ((ByteBuf) msg).readableBytes();
-		}
+				dataSentTime = System.nanoTime();
+			}
 
-		if (msg instanceof LastHttpContent) {
-			promise.addListener(future -> {
-				ChannelOperations<?, ?> channelOps = ChannelOperations.get(ctx.channel());
-				if (channelOps instanceof HttpServerOperations) {
-					HttpServerOperations ops = (HttpServerOperations) channelOps;
-					recordWrite(ops, uriTagValue == null ? ops.path : uriTagValue.apply(ops.path),
-							ops.method().name(), ops.status().codeAsText().toString());
-					if (!ops.isHttp2() && ops.hostAddress() != null) {
-						// This metric is not applicable for HTTP/2
-						// ops.hostAddress() == null when request decoding failed, in this case
-						// we do not report active connection, so we do not report inactive connection
-						recordInactiveConnection(ops);
+			dataSent += extractProcessedDataFromBuffer(msg);
+
+			if (msg instanceof LastHttpContent) {
+				promise.addListener(future -> {
+					try {
+						withHttpServerOperations(ctx, ops -> {
+							recordWrite(ops, uriTagValue == null ? ops.path : uriTagValue.apply(ops.path),
+									ops.method().name(), ops.status().codeAsText().toString());
+							if (!ops.isHttp2() && ops.hostAddress() != null) {
+								// This metric is not applicable for HTTP/2
+								// ops.hostAddress() == null when request decoding failed, in this case
+								// we do not report active connection, so we do not report inactive connection
+								recordInactiveConnection(ops);
+							}
+						});
+					} catch (RuntimeException e) {
+						log.warn("Exception caught while recording metrics.", e);
+						// Allow request-response exchange to continue, unaffected by metrics problem
 					}
-				}
 
-				dataSent = 0;
-			});
+					dataSent = 0;
+				});
+			}
+		} catch (RuntimeException e) {
```

Comment:
Do we need this `try/catch`? Is it possible to have a `RuntimeException` not related to metrics?

Reply:
No, also the actual handling is in a listener I likely only discovered later.